src/rp2_common/pico_stdio_usb/stdio_usb.c

```diff
@@ -127,19 +125,28 @@ static void stdio_usb_out_chars(const char *buf, int length) {
 
 int stdio_usb_in_chars(char *buf, int length) {
     uint32_t owner;
-    if (!mutex_try_enter(&stdio_usb_mutex, &owner)) {
-        if (owner == get_core_num()) return PICO_ERROR_NO_DATA; // would deadlock otherwise
-        mutex_enter_blocking(&stdio_usb_mutex);
-    }
+    // note we perform this check outside the lock, to try and prevent possible deadlock conditions
+    // with printf in IRQs (which we will escape through timeouts elsewhere, but that would be less graceful).
+    //
+    // these are just checks of state, so we can call them while not holding the lock.
+    // they may be wrong, but only if we are in the middle of a tud_task call, in which case at worst
+    // we will mistakenly think we have data available when we do not (we will check again), or
+    // tud_task will complete running and we will check the right values the next time.
+    //
     int rc = PICO_ERROR_NO_DATA;
-    if (stdio_usb_connected() && tud_cdc_available()) {
-        int count = (int) tud_cdc_read(buf, (uint32_t) length);
-        rc =  count ? count : PICO_ERROR_NO_DATA;
-    } else {
-        // because our mutex use may starve out the background task, run tud_task here (we own the mutex)
-        tud_task();
-    }
-    mutex_exit(&stdio_usb_mutex);
+//    if (stdio_usb_connected() && tud_cdc_available()) {
```

Comment:
did you mean to leave this code commented out

Reply:
ah - I see the second commit now

Reply:
no, but i had already fixed that!